From 0c6155824020ae0b1fbb658339554aa1953bb04f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 26 Aug 2022 13:46:20 +0300
Subject: [PATCH] [SW-86869] common: fix progress bit for boot desc

The start progress bit of boot desc progress was changed but not
updated in boot_desc_init function, so fix it.

Change-Id: I76fffd3e62360c22173b5b620ca45ce4a9b33b04
---

diff --git a/arch/gaudi2/arc/progress_bits.h b/arch/gaudi2/arc/progress_bits.h
index c4c7d63..a10f641 100644
--- a/arch/gaudi2/arc/progress_bits.h
+++ b/arch/gaudi2/arc/progress_bits.h
@@ -7,105 +7,107 @@
 	PROGRESS_INIT = 0,
 	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
 	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
-	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
-	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
-	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
-	PROGRESS_REVOKE_LOC = 15,
-	PROGRESS_REVOKE_LOC_0 = 16,
-	PROGRESS_REVOKE_LOC_1 = 17,
-	PROGRESS_KEY_LOC = 18,
-	PROGRESS_KEY_LOC_0 = 19,
-	PROGRESS_KEY_LOC_1 = 20,
-	PROGRESS_VER_LOC = 21,
-	PROGRESS_VER_LOC_0 = 22,
-	PROGRESS_VER_LOC_1 = 23,
-	PROGRESS_BRINGUP_PCI_PRI = 24,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 25,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 26,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 27,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 28,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 29,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 30,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 31,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 32,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 33,
-	PROGRESS_BRINGUP_PCI_SEC = 34,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 35,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 36,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 37,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 38,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 39,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 40,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 41,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 42,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 43,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 44,
-	PROGRESS_PPBOOT_START = 45,
-	PROGRESS_PPBOOT_FINISH = 46,
-	PROGRESS_BRINGUP_RAM_PRI = 47,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 48,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 49,
-	PROGRESS_BRINGUP_RAM_SEC = 50,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 51,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 52,
-	PROGRESS_ARCBOOT_RAM_START = 53,
-	PROGRESS_ARCBOOT_RECOVERY = 54,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 55,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 56,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 57,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 58,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 59,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 60,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 61,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 62,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 63,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 64,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 65,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 66,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 67,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 68,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 69,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 70,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 71,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 72,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 73,
-	PROGRESS_ARCBOOT_EROM_PCIE = 74,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 75,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 76,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 77,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 78,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 79,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 80,
-	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 81,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 82,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 83,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 84,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 85,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 86,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 87,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 88,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 89,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 90,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 91,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 92,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 93,
-	PROGRESS_ARCBOOT_EROM_TPM = 94,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 95,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 96,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 97,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 98,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 99,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 100,
-	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 101,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 102,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 103,
+	PROGRESS_INIT_SPI_BOOT_DESC_EXCEEDS_MAX_SIZE = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 14,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 15,
+	PROGRESS_REVOKE_LOC = 16,
+	PROGRESS_REVOKE_LOC_0 = 17,
+	PROGRESS_REVOKE_LOC_1 = 18,
+	PROGRESS_KEY_LOC = 19,
+	PROGRESS_KEY_LOC_0 = 20,
+	PROGRESS_KEY_LOC_1 = 21,
+	PROGRESS_VER_LOC = 22,
+	PROGRESS_VER_LOC_0 = 23,
+	PROGRESS_VER_LOC_1 = 24,
+	PROGRESS_BRINGUP_PCI_PRI = 25,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 26,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 27,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 28,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 29,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 30,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 31,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 32,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 33,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 34,
+	PROGRESS_BRINGUP_PCI_SEC = 35,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 36,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 37,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 38,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 39,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 40,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 41,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 42,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 43,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 44,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 45,
+	PROGRESS_PPBOOT_START = 46,
+	PROGRESS_PPBOOT_FINISH = 47,
+	PROGRESS_BRINGUP_RAM_PRI = 48,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 49,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 50,
+	PROGRESS_BRINGUP_RAM_SEC = 51,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 52,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 53,
+	PROGRESS_ARCBOOT_RAM_START = 54,
+	PROGRESS_ARCBOOT_RECOVERY = 55,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 56,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 57,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 58,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 59,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 60,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 61,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 62,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 63,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 64,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 65,
+	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 66,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 67,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 68,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 69,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 70,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 71,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 72,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 73,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 74,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 75,
+	PROGRESS_ARCBOOT_EROM_PCIE = 76,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 77,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 78,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 79,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 80,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 81,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 82,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 83,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 84,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 85,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 86,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 87,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 88,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 89,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 90,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 91,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 92,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 93,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 94,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 95,
+	PROGRESS_ARCBOOT_EROM_TPM = 96,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 97,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 98,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 99,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 100,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 101,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 102,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 103,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 104,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 105,
 };
diff --git a/arch/greco/arc_elf32/progress_bits.h b/arch/greco/arc_elf32/progress_bits.h
index c4c7d63..6101c50 100644
--- a/arch/greco/arc_elf32/progress_bits.h
+++ b/arch/greco/arc_elf32/progress_bits.h
@@ -7,105 +7,107 @@
 	PROGRESS_INIT = 0,
 	PROGRESS_INIT_SP_SRAM_NOT_COMPLETED = 1,
 	PROGRESS_INIT_ARC1_ICCM_INIT_FAILED = 2,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 3,
-	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 4,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 5,
-	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 6,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 7,
-	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 8,
-	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 9,
-	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 10,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 11,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 12,
-	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 13,
-	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 14,
-	PROGRESS_REVOKE_LOC = 15,
-	PROGRESS_REVOKE_LOC_0 = 16,
-	PROGRESS_REVOKE_LOC_1 = 17,
-	PROGRESS_KEY_LOC = 18,
-	PROGRESS_KEY_LOC_0 = 19,
-	PROGRESS_KEY_LOC_1 = 20,
-	PROGRESS_VER_LOC = 21,
-	PROGRESS_VER_LOC_0 = 22,
-	PROGRESS_VER_LOC_1 = 23,
-	PROGRESS_BRINGUP_PCI_PRI = 24,
-	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 25,
-	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 26,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 27,
-	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 28,
-	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 29,
-	PROGRESS_BRINGUP_PCI_PRI_DMA = 30,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 31,
-	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 32,
-	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 33,
-	PROGRESS_BRINGUP_PCI_SEC = 34,
-	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 35,
-	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 36,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 37,
-	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 38,
-	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 39,
-	PROGRESS_BRINGUP_PCI_SEC_DMA = 40,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 41,
-	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 42,
-	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 43,
-	PROGRESS_ARCBOOT_IN_RECOVERY = 44,
-	PROGRESS_PPBOOT_START = 45,
-	PROGRESS_PPBOOT_FINISH = 46,
-	PROGRESS_BRINGUP_RAM_PRI = 47,
-	PROGRESS_BRINGUP_RAM_PRI_DMA = 48,
-	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 49,
-	PROGRESS_BRINGUP_RAM_SEC = 50,
-	PROGRESS_BRINGUP_RAM_SEC_DMA = 51,
-	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 52,
-	PROGRESS_ARCBOOT_RAM_START = 53,
-	PROGRESS_ARCBOOT_RECOVERY = 54,
-	PROGRESS_ARCBOOT_RECOVERY_DMA = 55,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 56,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 57,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 58,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 59,
-	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 60,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 61,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 62,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 63,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 64,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 65,
-	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 66,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 67,
-	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 68,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 69,
-	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 70,
-	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 71,
-	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 72,
-	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 73,
-	PROGRESS_ARCBOOT_EROM_PCIE = 74,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 75,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 76,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 77,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 78,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 79,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 80,
-	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 81,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 82,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 83,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 84,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 85,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 86,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 87,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 88,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 89,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 90,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 91,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 92,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 93,
-	PROGRESS_ARCBOOT_EROM_TPM = 94,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 95,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 96,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 97,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 98,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 99,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 100,
-	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 101,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 102,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 103,
+	PROGRESS_INIT_SPI_BOOT_DESC_EXCEEDS_MAX_SIZE = 3,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_START = 4,
+	PROGRESS_INIT_SPI_BOOT_DESC_CRC_FAIL = 5,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_DEST = 6,
+	PROGRESS_INIT_SPI_BOOT_DESC_UNALIGNED_IMG_SIZE = 7,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_ERROR = 8,
+	PROGRESS_INIT_SPI_BOOT_DESC_DMA_FINISH = 9,
+	PROGRESS_INIT_SPI_BOOT_DESC_HDR_MAGIC_FAIL = 10,
+	PROGRESS_INIT_SPI_BOOT_DESC_VERIFICATION_FAIL = 11,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_SET_FAIL = 12,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_SET_FAIL = 13,
+	PROGRESS_INIT_SPI_BOOT_DESC_INFO_GET_FAIL = 14,
+	PROGRESS_INIT_SPI_BOOT_DESC_DATA_GET_FAIL = 15,
+	PROGRESS_REVOKE_LOC = 16,
+	PROGRESS_REVOKE_LOC_0 = 17,
+	PROGRESS_REVOKE_LOC_1 = 18,
+	PROGRESS_KEY_LOC = 19,
+	PROGRESS_KEY_LOC_0 = 20,
+	PROGRESS_KEY_LOC_1 = 21,
+	PROGRESS_VER_LOC = 22,
+	PROGRESS_VER_LOC_0 = 23,
+	PROGRESS_VER_LOC_1 = 24,
+	PROGRESS_BRINGUP_PCI_PRI = 25,
+	PROGRESS_BRINGUP_PCI_PRI_TOO_BIG = 26,
+	PROGRESS_BRINGUP_PCI_PRI_CRC_FAIL = 27,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_DEST_ADDR = 28,
+	PROGRESS_BRINGUP_PCI_PRI_UNALIGNED_IMG_SIZE = 29,
+	PROGRESS_BRINGUP_PCI_PRI_DMA_ERROR = 30,
+	PROGRESS_BRINGUP_PCI_PRI_DMA = 31,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_REVOKED = 32,
+	PROGRESS_BRINGUP_PCI_PRI_KEY_AUTH_FAIL = 33,
+	PROGRESS_BRINGUP_PCI_PRI_SECURITY = 34,
+	PROGRESS_BRINGUP_PCI_SEC = 35,
+	PROGRESS_BRINGUP_PCI_SEC_TOO_BIG = 36,
+	PROGRESS_BRINGUP_PCI_SEC_CRC_FAIL = 37,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_DEST_ADDR = 38,
+	PROGRESS_BRINGUP_PCI_SEC_UNALIGNED_IMG_SIZE = 39,
+	PROGRESS_BRINGUP_PCI_SEC_DMA_ERROR = 40,
+	PROGRESS_BRINGUP_PCI_SEC_DMA = 41,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_REVOKED = 42,
+	PROGRESS_BRINGUP_PCI_SEC_KEY_AUTH_FAIL = 43,
+	PROGRESS_BRINGUP_PCI_SEC_SECURITY = 44,
+	PROGRESS_ARCBOOT_IN_RECOVERY = 45,
+	PROGRESS_PPBOOT_START = 46,
+	PROGRESS_PPBOOT_FINISH = 47,
+	PROGRESS_BRINGUP_RAM_PRI = 48,
+	PROGRESS_BRINGUP_RAM_PRI_DMA = 49,
+	PROGRESS_BRINGUP_RAM_PRI_SECURITY = 50,
+	PROGRESS_BRINGUP_RAM_SEC = 51,
+	PROGRESS_BRINGUP_RAM_SEC_DMA = 52,
+	PROGRESS_BRINGUP_RAM_SEC_SECURITY = 53,
+	PROGRESS_ARCBOOT_RAM_START = 54,
+	PROGRESS_ARCBOOT_RECOVERY = 55,
+	PROGRESS_ARCBOOT_RECOVERY_DMA = 56,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 57,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 58,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 59,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 60,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 61,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 62,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 63,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 64,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 65,
+	PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE = 66,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_START = 67,
+	PROGRESS_INIT_EROM_BOOT_DESC_CRC_FAIL = 68,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_DEST = 69,
+	PROGRESS_INIT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 70,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_ERROR = 71,
+	PROGRESS_INIT_EROM_BOOT_DESC_DMA_FINISH = 72,
+	PROGRESS_INIT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 73,
+	PROGRESS_INIT_EROM_BOOT_DESC_VERIFICATION_FAIL = 74,
+	PROGRESS_INIT_EROM_BOOT_DESC_FAIL = 75,
+	PROGRESS_ARCBOOT_EROM_PCIE = 76,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 77,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 78,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 79,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 80,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 81,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 82,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 83,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 84,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 85,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 86,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 87,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 88,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 89,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 90,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 91,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 92,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 93,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 94,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 95,
+	PROGRESS_ARCBOOT_EROM_TPM = 96,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 97,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAIL101 = 98,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_D2ST = 99,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 100,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 101,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 102,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 103,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 104,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 105,
 };
diff --git a/common/boot_desc.c b/common/boot_desc.c
index 885f13d..91b9f97 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -70,7 +70,7 @@
 {
 	uint32_t i, img_ptrs_size;
 	uint32_t *p_offset = NULL;
-	enum arcboot_rom_progress dma_sts_base = PROGRESS_INIT_SPI_BOOT_DESC_DMA_START;
+	enum arcboot_rom_progress dma_sts_base = PROGRESS_INIT_SPI_BOOT_DESC_EXCEEDS_MAX_SIZE;
 	int err;
 	uint32_t copy_sz;
 	static struct hl_fw_img_desc *p_img_desc;
@@ -88,7 +88,7 @@
 		return p_boot_ptr->count;
 
 	if (img == HL_FW_BOOT_PTR_EROM) {
-		dma_sts_base = PROGRESS_INIT_EROM_BOOT_DESC_DMA_START;
+		dma_sts_base = PROGRESS_INIT_EROM_BOOT_DESC_EXCEEDS_MAX_SIZE;
 	} else {
 		if (arcboot_rom_global.validation_pass)
 			goto skip_boot_desc_process;
